tokio-util/Cargo.toml

```diff
@@ -27,7 +27,7 @@ net = ["tokio/net"]
 compat = ["futures-io"]
 codec = []
 time = ["tokio/time", "slab"]
-io = []
+io = ["tokio/rt"]
```

Comment:
mm. I'd like to avoid this dependence .... could we have Tokio expose no-op methods without rt?

Reply:
I will open another PR for this change.

I search the internal usage of `coop` under the tokio crate, there are many things are heavily using the `cfg_coop` and `poll_proceed_and_make_progress`. It is reasonable to expose an no-op method, and also writing a guide for downstream.

https://github.com/tokio-rs/tokio/blob/eb99e476e674f81095e2f8f15cbc92f8650f5364/tokio/src/io/util/mod.rs#L90-L102